### DISCUSSION TITLE
How to profile next dev compilation

### DISCUSSION THREAD

**Discussion by thoo1** [vercel/next.js#26704] [Help]
I am trying to debug an next v10 app. I am finding that next dev takes a long time and am trying to figure out why, what steps webpack is doing and spending time in to build. How do I do this?

**Comment by timneutkens** [comment]
Make sure you're using webpack 5, then you can use: https://github.com/vercel/next.js/issues/26588#issuecomment-868525614

**Reply by thoo1** [reply]
thanks, I got a lot more logs using webpack 5 and NEXT_WEBPACK_LOGGING=1

However we are having some errors building with webpack 5 (probably some webpack plugin we are using has a bug)
Is there any alternative for projects using webpack 4 still? Thanks!

**Reply by timneutkens** [reply]
There is not as we added this in webpack 5. Overall if you're seeing slowdowns it's highly recommended to use (now default) webpack 5 on Next.js 11 as we've been making a ton of improvements to dev speed.
